cuda_phase-retrieval/phase_algo.hpp

```diff
@@ -298,7 +419,9 @@ __global__ void init_random(double seed, curandState_t *states, int dimension)
     }
 }
 
-//sample random phase using array of random from input
+/**
+ * Sample random phase using array of random from input
+ */
 __global__ void random_phase(double *random, cufftDoubleComplex *y_hat, double *ptrMag, int dimension ) 
```

Comment:
Please add also useful documentation direct to the functions, otherwise `doxygen` is not able to generate useful documentation. For example (the doc strings are just examples - I have no idea if it is correct):

```c++
/**
 * \brief Sample random phase using array of random from input
 * \param random array of random numbers with the size of dimension
 * \param y_hat working data with size of dimension
 * \param ptrMag no idea
 * \param dimension size of all arrays
 * \return nothing, is a CUDA kernel ;-)
 */
__global__ void random_phase(double *random, cufftDoubleComplex *y_hat, double *ptrMag, int dimension )
```